react-components/src/components/BundleProductOptions/bundleProductOptions.js

```diff
@@ -36,7 +36,7 @@ const BundleProductOptions = () => {
     const productId = document.querySelector('[data-cmp-is=product]')?.id;
     const [bundleState, setBundleState] = useState(null);
     const intl = useIntl();
-    let { sku, showAddToWishList } = document.querySelector(bundleProductOptionsContainer)?.dataset || {};
+    let { sku, showAddToWishList, useUid } = document.querySelector(bundleProductOptionsContainer)?.dataset || {};
```

Comment:
useUid will be initialised with `''` which later evaluates to false as an empty string is a falsy value. This should be checked for undefined to cast it to a proper boolean

```
useUid = typeof useUid !== 'undefined'
```